File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();

        int curr = 0,
            zc = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i]) {
                nums[curr] = nums[i];
                curr++;
            } else {
                zc += 1;
            }
        }
        for (; curr < n; curr++) {
            nums[curr] = 0;
        }
    }
};

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        N = len(nums)

        z, nz = 0, 0
        while z < N and nz < N:
            if nums[nz] == 0:
                nz += 1
            elif nums[z] != 0:
                z += 1
            else:
                if nz > z:
                    nums[nz], nums[z] = nums[z], nums[nz]
                    z += 1
                    nz += 1
                else:
                    nz += 1